### ISSUE TITLE
Live tiles not working on Windows 11 Build 25398 (a.k.a windows server 23H2 RTM)

### ISSUE THREAD

**Issue by gailium119** [valinet/ExplorerPatcher#1621]
Client build of 25398 can be found here:
https://archive.org/details/25398.1.230610-1204.zn-release-clientmulti-x-64fre-en-us
25398 lacks StartUI.dll, can be found in PPIPro editionpack, uploading one here
[StartUI.zip](https://github.com/valinet/ExplorerPatcher/files/11987310/StartUI.zip)
Using 22000's StartTileData.dll causes StartMenuExperienceHost.exe crash with error code c0000005 in StartTileDataLegacy!DataStoreCache::TileStoreTransformer::TileContext::EnsureAppTypeAndCanElevate+0x1b3
Using 25398's original StartTileData.dll causes Live tiles failing to work.
Could anyone fix this?

**Comment by pyrates999** [comment]
That is an insider build and not supported because microsoft doesn't make symbols available to download for it: https://github.com/valinet/ExplorerPatcher/wiki/Symbols

**Comment by gailium119** [comment]
symbols are available for this build.

**Comment by pyrates999** [comment]
Is it a dev build?

**Comment by Vipin2003** [comment]
It is issue #1108 .... Follow #1591 for the fix... (Pin to Start is broken currrntly)

**Comment by gailium119** [comment]
> It is issue #1108 .... Follow #1591 for the fix... (Pin to Start is broken currrntly)

Does it still need some hooking when using the original files?

**Comment by Amrsatrio** [comment]
Live tiles also do not work on my 22621.2361 laptop whose installation began with 22000. But it does on my 22621.2283 desktop whose installation began with Windows 10 17763 (~2018). I guess it works only if the OS was upgraded from Windows 10?

My laptop:
![image](https://github.com/valinet/ExplorerPatcher/assets/20662640/e187abec-fd4b-46f6-9f74-ca4124915eb5)
My desktop:
![image](https://github.com/valinet/ExplorerPatcher/assets/20662640/c8c2f6d1-6049-4563-b989-af059eb687f2)

**Comment by Amrsatrio** [comment]
Closing this because live tiles work on latest 24H2 builds. With latest prerelease EP, you just need to drop StartUI.dll in the StartMenuExperienceHost directory.

**Comment by XiaoTDM** [comment]
@pyrates999 
Hi there! I would like to ask, I want the live tile continues to work while I try some methods, including but not limited to replace/patching files.

I have the following three questions:
First, is there a solution available at the moment?
Second, what is the version of Windows 11 for the last known live tile to work?
Third, what files or DLLs are related to the live tile update?

I'll try patching the file in the virtual machine, and if it works, I'll go to the physical machine to test further until it stabilizes

**Comment by milestprower92** [comment]
@XiaoTDM You can use [Live Tiles Anywhere](https://apps.microsoft.com/detail/9nr7qqk712pl?hl=en-US&gl=US)
to refresh the live tile database
which in most cases makes them work again

**Comment by XiaoTDM** [comment]
> @XiaoTDM You can use [Live Tiles Anywhere](https://apps.microsoft.com/detail/9nr7qqk712pl?hl=en-US&gl=US)
> to refresh the live tile database
> which in most cases makes them work again

I have tried but no effect. Weather, Sports and Money still do not update tiles

**Comment by TK50P** [comment]
<img width="1920" height="1080" alt="Image" src="https://github.com/user-attachments/assets/fb06413f-7a80-4969-8aa4-7506f98830a7" />
Seems to be working as of with latest cumulative updates.

Please lock this thread if you've forgot to lock.

**Comment by XiaoTDM** [comment]
Tried using 22H2's wpn and starttiledata.dll related file substitution, and the result is that explorer starts crashing infinitely.
Maybe I'll remove starttiledata.dll and try again

**Comment by XiaoTDM** [comment]
Based on my tests, directly replacing all the files will cause notifications from some software (phone connection) to not display. After further testing, simply replacing certain files can restore some live tile functionality:
`wpncore.dll` v10.0.22621.2070
Next, I will try to conduct a stability test, which will take some more time.

**Comment by XiaoTDM** [comment]
> Based on my tests, directly replacing all the files will cause notifications from some software (phone connection) to not display. After further testing, simply replacing certain files can restore some live tile functionality: `wpncore.dll` v10.0.22621.2070 Next, I will try to conduct a stability test, which will take some more time.

@pyrates999 Can we put this into the patch?

**Comment by pyrates999** [comment]
That's up to the developers here.

**Comment by Amrsatrio** [comment]
So just replacing wpncore.dll to that version?

In my testing with stock files on 26100.6899 and 28000.1, Weather and News live tiles work only after you open the respective apps.

If you want me to add the patch into EP, sorry to say but due to limited injection capabilities and the fact that wpncore.dll is loaded by svchost, at best I can only make a Windhawk mod for doing it.

**Comment by XiaoTDM** [comment]
> So just replacing wpncore.dll to that version?
> 
> In my testing with stock files on 26100.6899 and 28000.1, Weather and News live tiles work only after you open the respective apps.
> 
> If you want me to add the patch into EP, sorry to say but due to limited injection capabilities and the fact that wpncore.dll is loaded by svchost, at best I can only make a Windhawk mod for doing it.

Yes, just this one file. However, the bug I found was that it caused the content of notifications (such as email) in some apps to not be displayed, only the word "new notification". But if I switch to a version that doesn't have this problem, the live tile will no longer work.

Then the weather and news does need to be manually opened once before it can be loaded, which may be normal because WPN does not have their caches(?).

<img width="470" height="646" alt="Image" src="https://github.com/user-attachments/assets/2a2b607f-a441-4ba6-8be3-3a9666b74c0a" />

**Comment by XiaoTDM** [comment]
> However, the bug I found was that it caused the content of notifications (such as email) in some apps to not be displayed, only the word "new notification".

Well, it's necessary to test if the latest Windows 10 build has this problem

**Comment by pyrates999** [comment]
I run Windows 10 22h2, and it supports live tiles, but I don't use them.  And I read that windows 10 21h2 ltsc does not support it.

**Comment by XiaoTDM** [comment]
> I run Windows 10 22h2, and it supports live tiles, but I don't use them.  And I read that windows 10 21h2 ltsc does not support it.

By the way, could you test that if Windows / Wino Mail has this problem on notifications?
And does Windows 10 22H2 has 'Send to my phone' context menu? If it has, does the sending notification works well?

**Comment by pyrates999** [comment]
I don't have windows mail installed, neither do I have phone installed, sorry.

**Comment by XiaoTDM** [comment]
> So just replacing wpncore.dll to that version?
> 
> In my testing with stock files on 26100.6899 and 28000.1, Weather and News live tiles work only after you open the respective apps.
> 
> If you want me to add the patch into EP, sorry to say but due to limited injection capabilities and the fact that wpncore.dll is loaded by svchost, at best I can only make a Windhawk mod for doing it.

(EN) Given the current situation, it is likely that Microsoft "handy" removed the relevant code of the live tile when adding new interfaces and functions to the wpncore.dll. In order to keep the new interface and re-add the live tile functionality, it may be necessary to decompile the old version of the wpncore.dll and find the appropriate code that controls the live tiles and add them to the mod, and then let the mod hook the Windows Push Notification User Service inject this code into it.
(CN, Original) 鉴于目前情况来看，微软很可能是在向 wpncore.dll 添加新的接口和函数时“顺手”删除了动态磁贴的相关代码。为了同时保留新的接口并重新添加动态磁贴功能，可能需要反编译旧版本的 wpncore.dll 并找到控制动态磁贴的相应代码并加入 mod 中，然后让 mod hook 一下 Windows Push Notification User Service 将这些代码注入其中。

**Comment by XiaoTDM** [comment]
(EN) In addition, it should be clarified that the live tile itself is not actually broken, it is the content push function that relies on the WPN, which is why some tiles can be updated in the new version (such as News, Photos, and some locally updated apps) and others are invalid (such as Weather and other apps that rely on push services). Of course, this also causes the scope of modifications to be extended beyond the explorer itself when restoring live tiles.
(CN, Original) 另外需要澄清的是，动态磁贴本身其实并未失效，失效的是依赖WPN的内容推送功能，这就是为什么在新版本出现一些磁贴可以更新（如资讯、照片以及一些本地更新的应用）而另外一些就失效（如天气等依赖推送服务的应用），其实应该说动态磁贴失效是因为微软改动了WPN，和开始菜单本身关系不大。当然这也导致要恢复动态磁贴时修改范围会超出explorer本身。

**Comment by XiaoTDM** [comment]
> Of course, this also causes the scope of modifications to be extended beyond the explorer itself when restoring live tiles.

If we want to continue discussing these, it is better to open a new issue or discussion.

**Comment by pyrates999** [comment]
You should open up a new github repo if you're gonna patch windows dll files directly, or patch them in memory.

**Comment by XiaoTDM** [comment]
> You should open up a new github repo if you're gonna patch windows dll files directly, or patch them in memory.

That's right. It's beyond what EP can do.

**Comment by Amrsatrio** [comment]
Me and @Tourniquet88 (thanks for collecting builds) have dug deeper. Apparently a new feature flag in 25231 named WNS_Optimize_Cta (ID: 40670583) is the change that caused this problem.

When swapping wpncore.dll on my 26200.8737 installation to 25227 and restarting WpnUserService_xxxxx, live tiles sprung back to life. When swapping to 25231, the breakage consistent with the current behavior in 26xxx builds is observed. After running `vivetool /disable /id:40670583` (with the 25231 wpncore.dll in place) and restarting the service again, the live tiles sprung back to life with the same behavior as 25227 and before.

The trigger I'm using to test is Spotify while logged in. Launching the app updates the live tiles. If the DLL has the correct live tile behavior, it updates the live tile with playlist covers properly. If not, then no live tile, just Spotify logo and the app name with its accent color background.

Give me some time to think how to patch this one without swapping system files or hijacking COM paths. Likely this would be a Windhawk mod though, much like this one: https://github.com/ramensoftware/windhawk-mods/blob/main/mods/expanded-clipboard.wh.cpp

Download links for relevant DLLs from Microsoft symbol servers:
25227/25228 (same .text section): https://msdl.microsoft.com/download/symbols/wpncore.dll/6B971412196000/wpncore.dll
25231/25235 (same .text section): https://msdl.microsoft.com/download/symbols/wpncore.dll/ADCE281A197000/wpncore.dll

**Comment by XiaoTDM** [comment]
> Give me some time to think how to patch this one without swapping system files or hijacking COM paths. Likely this would be a Windhawk mod though, much like this one: https://github.com/ramensoftware/windhawk-mods/blob/main/mods/expanded-clipboard.wh.cpp

Thanks for your research, this could be a tough job because you need to make sure the existing notifications work while also preventing the tiles from being blocked.

**Comment by Amrsatrio** [comment]
The patches that I will make will only patch the code affected by that feature flag and keep other code intact. That's the least intrusive way to overcome this problem.

**Comment by XiaoTDM** [comment]
Can I ask more, what changes did Microsoft make to it? Since the feature flag name includes 'Optimize', does that mean Microsoft thought live-tile-related features were redundant code and decided to remove them? If that's the case, fixing it should be relatively simple; but if it also involves some of the new notification features, it would be trickier.

**Comment by Amrsatrio** [comment]
The feature flag in question refactored some code in `CtaManager` and `CtaHelpers`. I don't know what they do just yet, need to dig further. Just bindiffed those binaries and saw that the said feature flag was added and the `Cta` stuff were updated. I edited my comment above to include download links for 25227 and 25231 wpncore DLLs, so if you're curious and have IDA Pro, you could open the DLL and find the xrefs to `wil::details::FeatureImpl<__WilFeatureTraits_Feature_WNS_Optimize_Cta>::__private_IsEnabled`.

In addition to Optimize_Cta, there are a lot more of other Optimize_ feature flags in these two builds.

<img width="3840" height="2160" alt="Image" src="https://github.com/user-attachments/assets/84f324f6-4fc1-4fdc-b10d-ba8fae38a24f" />

**Comment by robbely** [comment]
I have 26200 on a bare metal system and on another device in a VM

Live tiles are working in the VM but not the bare metal system.
I tried "vivetool /disable /id:40670583" and restarting WpnUserService_xxxxx, (also tried rebooting) - live tiles still aren't working.
I tried downgrading wpcncore.dll to 25227 (8) from the link above (with service restarts and reboots) but still no success.
The live tiles in question are the News and Weather ones, they were both working before a recent windows update.

(EDIT: I also tried the 25231 (5) dll with the vivetool command - but still no luck)

Is there anything else I can try?

**Comment by XiaoTDM** [comment]
There may be an issue with the WPN database.

You can try to restore the database by backing up and deleting all the files in `%LocalAppData%\Microsoft\Windows\Notifications` after stopping all WPN services.

This action will delete all the live tiles and notifications, so please consider this carefully.

**Comment by robbely** [comment]
> You can try to restore the database

Many thanks. I tried this, restarted the services and reran News and Weather, but unfortunately it didn't work. I'm wondering if the issue is unrelated to push notifications, as I have now tried "Live Tiles Anywhere" and that app's tiles (News and Weather again) seem to work fine (but still not in the start menu).

**Comment by XiaoTDM** [comment]
> Many thanks. I tried this, restarted the services and reran News and Weather, but unfortunately it didn't work. 

Have you tried opening apps and leaving them for a while? Sometimes this is needed to refresh tiles.

If it still doesn't work, reset or reinstall the apps and try again.

**Comment by robbely** [comment]
I had already tried resetting and reinstalling which unfortunately didn’t work.

I tried running News and Weather while github was down (for an hour or so) but still no luck.

**Comment by XiaoTDM** [comment]
Then I might be at a loss for ideas. 

You can try what Microsoft officials often suggest: 

Try creating a new user and see if the lives work properly. If that doesn't work, use
`sfc /scannow`
`dism /online /cleanup-image /scanhealth`
`dism /online /cleanup-image /checkhealth`
`dism /online /cleanup-image /restorehealth`
to repair your system files.
EP files may be deleted and you may need to reinstall.

**Comment by robbely** [comment]
Many thanks for your help - I'll let you know if I have any success.

In the mean time I'd appreciate if anyone else has any suggestions.


UPDATE: Windows 11 got an update today and now the live tiles are working again :) - so must have been a windows glitch
